### PR TITLE
Fix: Algo Transfer and ASA Transfer execute

### PR DESCRIPTION
**Pull request by rmeena840**
Fix: Algo Transfer and ASA Transfer execute

### CODE REVIEW

File: dapp/src/utility/proposal.utility.ts [hunk 1]
```diff
@@ -250,3 +268,52 @@ export const checkProposalResult = (record: ProposalTableData) => {
...
 };
+
+/**
+ * Opt ASA to given account
+ * @param from : Account address that wants to opt asset
+ * @param assetID : Asset ID
+ * @param lsig : Logic Signature
+ * @param webMode : webmode instace to execute transaction
+ */
+export const optInASAToAccount = async (from: string, assetID: number, lsig: LogicSigAccount, webMode: any) => {
```
Comment (sczembor) [inline dapp/src/utility/proposal.utility.ts]: I think it's the other way around. We opt in the account to a specific asa. Lets rename it

File: dapp/src/utility/sigmadao.utility.ts [hunk 2]
```diff
@@ -37,3 +38,35 @@ export const validateFundAmount = (
...
+			`Please provide valid address.`
+		);
+	}
+};
+
+/**
+ * Verify Algorand address
+ * @param addr Address to be verifited
+ */
+export const isValidAlgroandAddress = (addr: string): boolean => {
```
Comment (sczembor) [inline dapp/src/utility/sigmadao.utility.ts]: I think we already have a function for it in algob, or we can just use one from algosdk

**Review by sczembor — COMMENTED** [review]
left few comments

File: dapp/src/utility/sigmadao.utility.ts (hunk 2 above)
Comment (rmeena840) [reply dapp/src/utility/sigmadao.utility.ts:67]: Yes, decodeAddress method is from algosdk. See the body of method.

Comment (rmeena840) [reply dapp/src/utility/sigmadao.utility.ts]: We can't use algob in frontend.

File: dapp/src/utility/proposal.utility.ts (hunk 1 above)
Comment (rmeena840) [reply dapp/src/utility/proposal.utility.ts]: Done

File: dapp/src/utility/sigmadao.utility.ts (hunk 2 above)
Comment (rmeena840) [reply dapp/src/utility/sigmadao.utility.ts]: Thx, I have used the method from algosdk.

**Comment by rmeena840** [comment]
@sczembor Ready for another review

File: dapp/src/utility/proposal.utility.ts [hunk 3]
```diff
@@ -250,3 +252,52 @@ export const checkProposalResult = (record: ProposalTableData) => {
...
+
+/**
+ * Transfer ASA from one account to another account
+ * @param from : Account address that wants to send transfer ASA
+ * @param to : Account address that wants to receive ASA
+ * @param numberOfASA : Number of ASA that has to be transferred
+ * @param assetID : Asset ID
+ * @param webMode : webmode instace to execute transaction
+ */
+export const transferASA = async (from: string, to: string, numberOfASA: number, assetID: number, webMode: any) => {
```
Comment (Megha-Dev-19) [inline dapp/src/utility/proposal.utility.ts]: I don't think we would require this function

File: dapp/src/utility/sigmadao.utility.ts [hunk 4]
```diff
@@ -37,3 +38,25 @@ export const validateFundAmount = (
 		return Promise.resolve();
 	}
 };
+
+/**
+ * Validate the Algorand Address
+ * @param _rule the antd rules
+ * @param addr Address
+ */
+export const validateAlgroandAddress = (
```
Comment (Megha-Dev-19) [inline dapp/src/utility/sigmadao.utility.ts]:
```suggestion
export const validateAlgorandAddress = (
```

**Review by Megha-Dev-19 — CHANGES_REQUESTED** [review]
You haven't added checks in create proposal 
To see if dao fund lsig has enough algos and asa to make that particular type of proposal

File: dapp/src/utility/proposal.utility.ts (hunk 3 above)
Comment (rmeena840) [reply dapp/src/utility/proposal.utility.ts]: Removed

File: dapp/src/pages/CreateProposal.vue [hunk 5]
```diff
@@ -387,12 +407,23 @@ export default defineComponent({
 						`int:${executeBefore}`, // execute_before
 						`int:${proposal_type}`, // type
 					];
+					// validate proposal parameters
+					if (
+						await this.validateProposalParam(
```
Comment (Megha-Dev-19) [inline dapp/src/pages/CreateProposal.vue]:
```suggestion
						await this.validateProposalParams(
```

File: dapp/src/pages/CreateProposal.vue [hunk 6]
```diff
@@ -465,6 +496,57 @@ export default defineComponent({
 		onFinishFailed(errorinfo: Event) {
 			console.warn("Failed:", errorinfo);
 		},
+		async validateProposalParam(
+			proposal_type: any,
```
Comment (Megha-Dev-19) [inline dapp/src/pages/CreateProposal.vue]: wouldn't it be a number?

File: dapp/src/pages/CreateProposal.vue [hunk 7]
```diff
@@ -465,6 +496,57 @@ export default defineComponent({
...
+			addr: string,
+			amount: number,
+			asaId: number
+		): Promise<boolean> {
+			return (
+				(proposal_type === ProposalType.ALGO_TRANSFER &&
+					!(await this.isDAOHavingRequiredFund(addr, amount))) ||
```
Comment (Megha-Dev-19) [inline dapp/src/pages/CreateProposal.vue:509]:
```suggestion
					!(await this.isDAOSufficientlyFunded(addr, amount))) ||
```

**Review by Megha-Dev-19 — APPROVED** [review]
Left Few suggestions, LGTM

File: dapp/src/pages/CreateProposal.vue (hunk 6 above)
Comment (rmeena840) [reply dapp/src/pages/CreateProposal.vue]: Created enum for this.

**Pull request merged**